src/Interpreters/Context.cpp

```diff
@@ -1661,7 +1661,12 @@ void Context::resetZooKeeper() const
 static void reloadZooKeeperIfChangedImpl(const ConfigurationPtr & config, const std::string & config_name, zkutil::ZooKeeperPtr & zk)
 {
     if (!zk || zk->configChanged(*config, config_name))
+    {
+        if (zk)
```

Comment:
BTW maybe it worth a comment in the code? (like this one https://github.com/ClickHouse/ClickHouse/pull/21264#issuecomment-786891304) Or not here but for `finalize()` method